### DISCUSSION TITLE
Set Env from response body html

### DISCUSSION THREAD

**Discussion by shawnferry** [thunderclient/thunder-client-support#898] [Q&A]
I think I'm missing something that should be simple.  

I need to pull a value from an initial html response prior to authenticating but I can't figure out where I'm going wrong. 

In a dev tools console I can get the value with 
`document.querySelector("form[action='/api-auth/login/'] > input[name=csrfmiddlewaretoken]").value` 

However, I can't figure out how to extract the same value from the html body.

**Comment by rangav** [comment]
Hi @shawnferry 

I think you should use custom filter and indexOf function

Custom filter - `text | getText`

Then in custom filter
```
var index1 = input.indexOf("middlewareToken")
Input = input.substring(index1)
var index2 = input.indexOf("</input>")
input = input.substring(0, index2)
console.log(input)
```

the above is rough idea how to extract text, I will research and see if there is any html library for extracting text.

**Comment by rangav** [comment]
Now you can import any Node module in Custom Filters, If you there is any node module for html parsing you can use it.

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v2.1.0

Please let me know your feedback.
